src/forms/form-fields/drop-down-async.js

```diff
@@ -265,6 +270,7 @@ DropDownAsync.propTypes = {
     model: PropTypes.object,
     options: PropTypes.array,
     onOptionsLoaded: PropTypes.func,
+    fieldFilter: PropTypes.string,
```

Comment:
This was added to prevent adding `programStageDataElements`-fields to every place `programs`-dropdown are used. 

This prop should've been added a long time ago, so we wouldn't need the "hacky" `fieldsForReferenceType` above.